src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs

```diff
@@ -119,19 +120,20 @@ public override void SetResponseRaw(HttpResponseMessage response, Stream stream,
             using (var parser = new UnmanagedJsonParser(context, state, "multi_get/response"))
             using (context.GetManagedBuffer(out JsonOperationContext.ManagedPinnedBuffer buffer))
             {
-                if (UnmanagedJsonParserHelper.Read(stream, parser, state, buffer) == false)
-                    ThrowInvalidResponse();
+                var peepingTomStream = new PeepingTomStream(stream);
```

Comment:
This generates a 4KB buffer on every MultIGet call, let's see if we can use the context to get a pinnable buffer that is reused between calls.